htdocs/core/lib/files.lib.php

```diff
@@ -1789,7 +1789,9 @@ function addFileIntoDatabaseIndex($dir, $file, $fullpathorig = '', $mode = 'uplo
 
 		if (is_object($object) && $object->id > 0) {
 			$ecmfile->src_object_id = $object->id;
-			$ecmfile->src_object_type = $object->table_element;
+			if (isset($object->table_element)) $ecmfile->src_object_type = $object->table_element;
```

Comment:
Setting src_object_id wihout setting src_object_type is useless.
For this reason table_element mus always be set. And if not set, we should report an error instead of allowing to record a data that we won't be able to reuse later.

Reply:
Thanks for your reply, I updated the PR.